adapters/esp8266_mock.h

```diff
@@ -0,0 +1,446 @@
+#ifndef _ESP8266_MOCK_H_
+#define _ESP8266_MOCK_H_
+
+#include <stdint.h>
+#include "azure_c_shared_utility/umock_c_prod.h"
+
+typedef void SSL;
+typedef void SSL_CTX;
+typedef void SSL_METHOD;
+
+/*
+ * SSL_CTX_new - create a SSL context
+ *
+ * @param method - the SSL context configuration file
+ *
+ * @return the context point, if create failed return NULL
+ */
+//SSL_CTX* SSL_CTX_new(SSL_METHOD *method);
+MOCKABLE_FUNCTION(, SSL_CTX*, SSL_CTX_new, SSL_METHOD*, method);
```

Comment:
Adding wrapper calls means more calling stack usage. Waiting to hear back from LCA. Unless LCA says we cannot mock these apis, we should not add wrapper calls.